src/awkward/_connect/jax/reducers.py

```diff
@@ -111,7 +111,16 @@ def apply(
         shifts: ak.index.Index | None,
         outlength: ShapeItem,
     ) -> ak.contents.NumpyArray:
-        raise RuntimeError("Cannot differentiate through count_zero")
+        assert isinstance(array, ak.contents.NumpyArray)
+        result = jax.numpy.ones_like(array.data, dtype=array.dtype)
+        result = jax.ops.segment_sum(result, parents.data)
+
+        if array.dtype.type in (np.complex128, np.complex64):
```

Comment:
I think we would be better to check "is this complex"
```suggestion
        if np.issubdtype(array.dtype, np.complexfloating):
```

Reply:
I see, thanks!